src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -404,7 +404,16 @@ private void validateRangePartitionTableWithInputConfiguration(Table table, Rang
 
   private void validateColumnForPartition(@Nullable String columnName, @Nullable Schema schema,
                                           FailureCollector collector) {
-    if (columnName == null || schema == null) {
+    if (containsMacro(NAME_PARTITION_BY_FIELD) || containsMacro(NAME_PARTITIONING_TYPE) || schema == null) {
+      return;
+    }
+    PartitionType partitioningType = getPartitioningType();
+    if (Strings.isNullOrEmpty(columnName)) {
+      if (partitioningType == PartitionType.INTEGER) {
```

Comment:
Is this check not require if partitioningType == PartitionType.TIME?

Reply:
Also, can we add a unit test for this in BigQuerySinkTest?

Reply:
BigQuery has 3 types of partitions: 

1. **Ingestion time:** Tables are partitioned based on the data's ingestion (load) time or arrival time.
2. **Date/timestamp/datetime:** Tables are partitioned based on a TIMESTAMP, DATE, or DATETIME column.
3. **Integer range:** Tables are partitioned based on an integer column.

https://cloud.google.com/bigquery/docs/partitioned-tables

In CDAP if for Time Partition if there is no field specified, the table will partition by ingestion time, so partition columnName can be empty.


We have added a unit test where partition type is time and the partition field is provided

Reply:
I see, thanks for explaining!